GL: make gl_skymode 2 more consistent

bkoropoff opened the pull request:
- Respect render_stretchsky setting
- Consistently correct scale for FOV
- Tweak behavior to be closer to software
- Simplify code and remove magic constants where possible
- Move the fussy math into a single function
-----

## Software

![doom00](https://github.com/kraflab/dsda-doom/assets/2101303/ea029db5-3413-4c13-b61c-856c3a3fecd3)

## GL, FOV 90, mouselook off

![doom01](https://github.com/kraflab/dsda-doom/assets/2101303/38d89911-5011-4eaa-93c4-bb1c640cc281)

## GL, FOV 90, mouselook on

![doom02](https://github.com/kraflab/dsda-doom/assets/2101303/4a8c1c19-a39e-4265-a94d-3ca3769054e0)

## GL, FOV 110, mouselook off

![doom03](https://github.com/kraflab/dsda-doom/assets/2101303/c7b30b02-7dc0-4782-a482-686629e5896c)

## GL, FOV110, mouselook on

![doom04](https://github.com/kraflab/dsda-doom/assets/2101303/9b8f1ffa-3069-428d-ac39-c912709470df)

## gl_skymode 3, mouselook off

![doom00](https://github.com/kraflab/dsda-doom/assets/2101303/f93ee962-75c4-4a85-9815-898a090686ec)

## gl_skymode 3, mouselook on

![doom01](https://github.com/kraflab/dsda-doom/assets/2101303/dbca12b8-14e3-4978-8546-2ca6461c6f1c)


Closes #373

bkoropoff commented:
Took a bit to untangle all of that logic, but GL behavior is very close to software in all cases and Heretic and Hexen look good too.

bkoropoff commented:
The line transfer handling code was inconsistent between mouselook on and off (?!), and I'm not sure I picked the right "version".  Is there a good test map for that stuff?

andrey-budko commented:
All these gl_skymode 3 stretched skies look weird to me (render_stretchsky is 1 by default). GZDoom should be the reference for skydome code, right?

fabiangreffrath commented:
> The line transfer handling code was inconsistent between mouselook on and off (?!), and I'm not sure I picked the right "version". Is there a good test map for that stuff?

If you mean sky transfers, MBF has a test map for this:
https://github.com/fabiangreffrath/woof/blob/master/examples/sky.wad

bkoropoff commented:
I can disable the stretching for mode 3.  The fact that it's conditional leads me to think it was supposed to be doing it, the logic was just broken.

bkoropoff commented:
Resolved issues with different tilings (based on image size), fixed/tested sky transfers against eviternity

kraflab commented:
Very nice simplification of this code 👌  
Going to test it out soon

kraflab commented:
skymode 2 is broken in heretic and hexen (heretic and hexen have tall skies and you can look upwards even without mouselook):
![doom97](https://github.com/kraflab/dsda-doom/assets/25529060/67e285fc-7035-4307-bebc-20ba493cfbee)
![doom98](https://github.com/kraflab/dsda-doom/assets/25529060/996e1240-91c1-4070-91a8-830d7e0fc2d8)

bkoropoff committed:
GL: make gl_skymode 2 more consistent

- Respect render_stretchsky setting
- Consistently correct scale for FOV
- Tweak behavior to be closer to software
- Simplify code and remove magic constants where possible
- Move the fussy math into a single function

bkoropoff commented:
Further refactored/cleaned up the core transform to make some semblance of sense.  It also fixed the Heretic problem.

bkoropoff commented:
Dropped the skymode 3 changes

kraflab commented:
Thanks 👍
